File: app/src/main/java/com/pleon/buyt/database/converter/QuantityUnitConverter.java
package com.pleon.buyt.database.converter;

import com.pleon.buyt.model.Quantity.Unit;

import androidx.room.TypeConverter;

public class QuantityUnitConverter {

    @TypeConverter
    public static Unit convertToUnit(String name) {
        return Unit.valueOf(name);
    }

    @TypeConverter
    public static String convertToName(Unit unit) {
        return unit.name();
    }
}

File: app/src/main/kotlin/com/pleon/buyt/database/converter/QuantityUnitConverter.kt
package com.pleon.buyt.database.converter

import androidx.room.TypeConverter
import com.pleon.buyt.model.Item.Quantity

class QuantityUnitConverter {

    @TypeConverter
    fun convertToUnit(name: String) = Quantity.Unit.valueOf(name)

    @TypeConverter
    fun convertToName(unit: Quantity.Unit) = unit.name
}
